app/src/main/java/uk/co/neuralcubes/neuralates/ControlFragment.java

```diff
@@ -139,6 +139,7 @@ public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l)
             setEnabledStateForViews(mMuseActions, true);
 
             if (mSphero.isPresent()) {
+                if (mController.isPresent()) mController.get().unlink();
```

Comment:
This line should probably go outside the outer if statement. Also, please add the curly brackets.